src/main/java/org/example/bookstore/repository/user/UserRepository.java

```diff
@@ -1,10 +1,14 @@
 package org.example.bookstore.repository.user;
 
+import java.util.Optional;
 import org.example.bookstore.model.User;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface UserRepository extends JpaRepository<User,Long> {
     boolean existsByEmail(String email);
+
+    Optional<User> findByEmail(@Param("email") String email);
```

Comment:
you can siplify that, just write Optional<User> findByEmail(String email) - its the same thing